src/transformers/trainer.py

```diff
@@ -572,6 +580,12 @@ def evaluate(
 
         output = self._prediction_loop(eval_dataloader, description="Evaluation")
 
+        logs = {}
+        for key, value in output.metrics.items():
+            eval_key = "eval_{}".format(key)
+            logs[eval_key] = value
+        self._log(logs)
```

Comment:
Evaluation metrics are now directly logged in `evaluate` so that we have access to final model metrics at the end of the run.